### PR TITLE
plugin/nodeattestor: Add agent and server sshpop plugins

### PR DESCRIPTION
**Pull request by APTy**
This implements the ssh handshake protocol introduced in #838 as server and agent plugins. 

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Usage**

agent.conf
```
    NodeAttestor "sshpop" {
        plugin_data {
            host_cert_path = "./conf/agent/dummy_agent_ssh_key-cert.pub"
	    host_key_path = "./conf/agent/dummy_agent_ssh_key"
        }
    }
```

server.conf
```
    NodeAttestor "sshpop" {
        plugin_data {
            cert_authorities = ["ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIEAWPAsKJ/qMYUIBeH7BLMRCE/bkUvMHX+7OZhANk45S"]
        }
    }
```

logs
```
$ ./cmd/spire-server/spire-server run
...
DEBU[0002] Signing CSR for Agent SVID spiffe://example.org/spire/agent/sshpop/SHA256:21Aic/muK032oJMhLfU1/CMNcGmfAnvESeuH5zyFw/g  subsystem_name=node_api
DEBU[0002] Signed x509 SVID "spiffe://example.org/spire/agent/sshpop/SHA256:21Aic/muK032oJMhLfU1/CMNcGmfAnvESeuH5zyFw/g" (expires 2019-04-22T18:15:27Z)  subsystem_name=ca_manager
```

### CODE REVIEW

File: pkg/agent/plugin/nodeattestor/sshpop/sshpop.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	return nil
+}
+
+// Configure configures the Plugin.
+func (p *Plugin) Configure(ctx context.Context, req *plugin.ConfigureRequest) (*plugin.ConfigureResponse, error) {
+	sshclient, err := sshpop.NewClient(req.GlobalConfig.GetTrustDomain(), req.Configuration)
+	if err != nil {
+		return nil, err
+	}
+	p.sshclient = sshclient
```
Comment (azdagron) [inline pkg/agent/plugin/nodeattestor/sshpop/sshpop.go]: the atomic boolean is not sufficient to protect against data races against `p.sshclient`. I'd recommend using a mutex to protect access to `p.sshclient`. A `nil` value for `p.sshclient` could be interpreted as "not configured".

File: pkg/server/plugin/nodeattestor/sshpop/sshpop.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	return nil
+}
+
+func (p *Plugin) Configure(ctx context.Context, req *plugin.ConfigureRequest) (*plugin.ConfigureResponse, error) {
+	sshserver, err := sshpop.NewServer(req.GlobalConfig.GetTrustDomain(), req.Configuration)
+	if err != nil {
+		return nil, err
+	}
+	p.sshserver = sshserver
+	p.configured.Set(true)
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/sshpop/sshpop.go]: same comment about data races here....

File: doc/plugin_server_nodeattestor_sshpop.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+The SPIFFE ID produced by the plugin is based on the certificate fingerprint,
+which is an unpadded base64 encoded sha256 hash of the certificate in openssh format.
+
+```
+spiffe://<trust-domain>/spire/agent/sshpop/<fingerprint>
+```
+
+| Configuration | Description | Default                 |
+| ------------- | ----------- | ----------------------- |
+| `cert_authorities` | A list of trusted CAs in ssh `authorized_keys` format. | |
```
Comment (azdagron) [inline doc/plugin_server_nodeattestor_sshpop.md]: wonder if we should also support a `cert_authories_file` option to load authorities from disk (could merge the two together if both specified)?

Comment (APTy) [reply doc/plugin_server_nodeattestor_sshpop.md]: I'm definitely open to adding that, but maybe in a subsequent PR as this one is already fairly big?

Comment (azdagron) [reply doc/plugin_server_nodeattestor_sshpop.md]: Works for me!

**Pull request merged**